SweetsFactory/src/app/shared/product/productItem.component.ts

```diff
@@ -1,13 +1,49 @@
 import { Component, Input } from '@angular/core';
 import { Product } from '../model/product';
+import {
+    trigger,
+    state,
+    style,
+    animate,
+    transition
+} from '@angular/animations';
 
 @Component({
     selector: 'product-item',
     templateUrl: './productItem.component.html',
-    styleUrls: ['./productItem.component.css']
+    styleUrls: ['./productItem.component.css'],
+    animations: [
+        trigger('prodItemState', [
+            state('inactive', style({
+                backgroundColor: 'transparent',
+                opacity: 1
+            })),
+            state('active', style({
+                top: 0,
+                backgroundColor: 'white',
+                opacity: 0.7
+            })),
+            transition('inactive => active', [
+                animate(250)
+            ]),
+            transition('active => inactive', [
+                animate(250)
+            ])
+        ])
+    ]
 })
 
 export class ProductItemComponent {
     @Input() product: Product;
-    constructor() { }
+    state = 'inactive';
```

Comment:
What is the purpose of  the "state" variable in this case?

Reply:
I switch trough states when animating the overlay for a product

Reply:
You don't need javascript logic for doing such animations.